### ISSUE TITLE
chart: add support for Ingress v1 API

### ISSUE THREAD

**Issue by matiashaller** [jenkins-x/jx-pipelines-visualizer#120]
Hi,

After running "jx dash" it returns:

W0520 01:15:15.674690   20991 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
error: no dashboard URL. Check you have 'chart: jx3/jx-pipelines-visualizer' in your helmfile.yaml

Checked out the file "[ingress.yaml](https://github.com/jenkins-x/jx-pipelines-visualizer/blob/master/charts/jx-pipelines-visualizer/templates/ingress.yaml)" has "networking.k8s.io/v1beta1", can you please change it to networking.k8s.io/v1.

Thank you
Matias.

**Comment by vbehar** [comment]
Hi,

we need to be careful upgrading to the v1 api for ingress, because it's been introduced in kube v1.19, and some people might be using an older kube version - see https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122

but it is just a warning and shouldn't block you from opening the dashboard URL. I'm thinking that there is another issue here. Can you try to list your ingresses to see if there is one for the pipelines-visualizer?
Feel free to come on the slack channel https://jenkins-x.io/community/#slack if you need help debugging this issue

**Comment by hervelemeur** [comment]
`helm create` now (?) add a test on Kubernetes version to set one or the other ingress:
```
{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
```
Maybe something like that could be done?

**Comment by vbehar** [comment]
yes, in fact it's a bit more complex when supporting the v1 API, because some fields have been renamed, as explained in https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122

FYI there is an example of ingress template that supports both v1beta1 and v1 APIs here: https://github.com/grafana/helm-charts/blob/main/charts/grafana/templates/ingress.yaml

**Comment by vbehar** [comment]
fixed by #134
